includes/payment-methods/class-upe-payment-method.php

```diff
@@ -102,6 +110,16 @@ public function is_reusable() {
 		return $this->is_reusable;
 	}
 
+	/**
+	 * Returns boolean dependent on whether payment method will accept charges
+	 * with chosen currency
+	 *
+	 * @return bool
+	 */
+	public function is_currency_valid() {
+		return empty( $this->currencies ) || in_array( get_woocommerce_currency(), $this->currencies, true );
```

Comment:
I included the global `get_woocommerce_currency()` here essentially so that this function could work with `array_filter` like the other similar payment method filters. I defend this choice, because this function is only called when creating a setup or payment intent, where this function will be used to return the currency anyways (as opposed to taking the currency from the order or elsewhere), but once again I'm open to alternatives if this approach is deemed too lazy.